File: job-board-ui/src/model/job-offer-application/JobOfferApplicationModel.ts
import EmployeeAccountModel from "../employee-account/EmployeeAccountModel";
import JobOfferApplicationState from "../enum/JobOfferApplicationState";

export default interface JobOfferApplicationModel {
    id: number,
    state: JobOfferApplicationState,
    employee: EmployeeAccountModel,
    offerId: number,
    applicationDate: string,
    offerTitle: string
}

File: JobBoard.Model/JobOfferApplication/JobOfferApplicationModel.cs
using JobBoard.Domain.Enum;
using JobBoard.Model.EmployeeAccount;

namespace JobBoard.Model.JobOfferApplication
{
    public class JobOfferApplicationModel
    {
        public long Id { get; set; }
        public DateTimeOffset ApplicationDate { get; set; }
        public JobOfferApplicationState State { get; set; }
        public EmployeeAccountModel Employee { get; set; }
        public long OfferId { get; set; }
        public string OfferTitle { get; set; }
    }
}
